avm/res/dev-test-lab/lab/main.bicep

```
@@ -226,95 +226,86 @@ resource lab_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock ?
 }
 
 module lab_virtualNetworks 'virtualnetwork/main.bicep' = [
-  for (virtualNetwork, index) in virtualnetworks: {
+  for (virtualNetwork, index) in (virtualnetworks ?? []): {
     name: '${uniqueString(deployment().name, location)}-Lab-VirtualNetwork-${index}'
     params: {
       labName: lab.name
       name: virtualNetwork.name
       tags: virtualNetwork.?tags ?? tags
       externalProviderResourceId: virtualNetwork.externalProviderResourceId
-      description: contains(virtualNetwork, 'description') ? virtualNetwork.description : ''
-      allowedSubnets: contains(virtualNetwork, 'allowedSubnets') ? virtualNetwork.allowedSubnets : []
-      subnetOverrides: contains(virtualNetwork, 'subnetOverrides') ? virtualNetwork.subnetOverrides : []
+      description: virtualNetwork.?description ?? ''
+      allowedSubnets: virtualNetwork.?allowedSubnets ?? []
+      subnetOverrides: virtualNetwork.?subnetOverrides ?? []
```

```suggestion
      description: virtualNetwork.?description
      allowedSubnets: virtualNetwork.?allowedSubnets
      subnetOverrides: virtualNetwork.?subnetOverrides
```

While on the subject of optimization & defaults - passing `null` into a linked module should automatically pull the implemented parameter's default

---

We can also merge the PR without that change (as I presume you wanted to address the watnings primarily). But I wanted to at least mention it in case you're curious :) Feel free to resolve if not convinced 😄

---

There's always a room for improvement 😎 updated and retested.